pyproject.toml

```diff
@@ -175,9 +177,18 @@ before-build = [
 HATCH_BUILD_HOOKS_ENABLE = "1"
 MYPYC_OPT_LEVEL = "3"
 MYPYC_DEBUG_LEVEL = "0"
+AIOHTTP_NO_EXTENSIONS = "1"
+
 # Black needs Clang to compile successfully on Linux.
 CC = "clang"
-AIOHTTP_NO_EXTENSIONS = "1"
+
+[tool.cibuildwheel.macos]
+build-frontend = { name = "build", args = ["--no-isolation"] }
+# Unfortunately, hatch doesn't respect MACOSX_DEPLOYMENT_TARGET
+before-build = [
+    "python -m pip install 'hatchling==1.18.0' hatch-vcs hatch-fancy-pypi-readme 'hatch-mypyc>=0.16.0' 'mypy==1.5.1' 'click==8.1.3'",
```

Comment:
We'll have to remember to update this pin when we upgrade mypy.

Also, any specific reason we need to pin click and not our other dependencies here?

Reply:
We pinned click because 8.1.4 and 8.1.5 broke mypyc builds. I copied all the pins here from elsewhere in this file, with the exception of hatchling because we patch it